npm/private/npm_translate_lock.bzl

```diff
@@ -499,6 +501,10 @@ def npm_translate_lock(
 
         pnpm_version: pnpm version to use when generating the @pnpm repository. Set to None to not create this repository.
 
+            Deprecated for bzlmod, use `pnpm_extension` instead.
+
+        pnpm_extension: label of the pnpm extension to use
```

Comment:
I think I might be getting bzlmod terminology confused here. Let me know if there's a better term or way of phrasing this...

Reply:
But if you are passing in a label, why not just make it the full label to a pnpm binary?

Reply:
Is "repo" the correct term and not "extension"?

Reply:
Yeah, the extension creates a "repo/workspace". This is a... "label to a repo"?.

Reply:
It's essentially the same as how repo rules create repos/workspaces when using WORKSPACE. Now they are just wrapped in module extensions and namespaced.

Reply:
But the full binary vs just the repo(extension?) name is what I'd like to discuss before merging, what the public API looks like...

Currently in this PR it assumes what you pass in is repo name representing the pnpm package, created by the pnpm extension. Then rules_js does `relative(":package/bin/pnpm.cjs")` to point to the specific file within the pnpm package. This means the public API is for the user to specify the pnpm repo name, not the precise .js file being executed. This means the "pnpm repo name" essentially must be a `npm_import` of the `pnpm` package (and I sure hope the pnpm version doesn't change that `bin/pnpm.cjs` path 😓).

Reply:
Pnpm could theoretically come from some other repository than the one defined by our bzlmod extension, so I would favour passing in the full label.

Reply:
If it's the full label then users could pass in whatever they want, which is a good thing, but it also means they need to understand the internal paths of pnpm :/

Reply:
Can you just default the label? Which will always be valid because we always register the pnpm toolchain in the root module?

Reply:
> I think if pnpm_repo would be clearer. I didn't even know you could have a label to just the repository without referencing a target.. 🤔

Sorry I missed this. It resolves to the default target, then within the rules_js repo it does `.relative` to change to the specific pnpm bin .js file. This is the part that is iffy imo and I want to discuss more.

> Can you just default the label? Which will always be valid because we always register the pnpm toolchain in the root module?

Doesn't that mean it gets defaulted to the label instantiated within the rules_js repo though? So would be scoped to the rules_js repo and not to the users repo. That's what the default value does to reference `@pnpm` within rules_js...